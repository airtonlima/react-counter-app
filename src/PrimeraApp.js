import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ 
    saludo, 
    subtitulo, 
    otra
}) => {

    const obj = {
        nombre: 'Airton',
        edad: 26
    };

    return (
        <React.Fragment>

            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
            {/* <p>{ otra }</p> */}

            <pre>{ JSON.stringify(obj, null, 3) }</pre>
            
        </React.Fragment>
    );
};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string,
    otra: PropTypes.number
};

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
};

export default PrimeraApp;
