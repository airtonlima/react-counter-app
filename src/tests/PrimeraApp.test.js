import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {
    
    // test('Debe de mostrar el mensaje Hola, soy Goku', () => {
    //     const saludo = 'Hola, soy Goku';
    //     const wrapper = render(<PrimeraApp saludo={ saludo } />);
    //     expect(wrapper.getByText(saludo)).toBeInTheDocument();
    // });

    test('#01 - Debe de mostrar <PrimeraApp /> correctamente', () => {
        
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> );

        expect(wrapper).toMatchSnapshot();
    });

    test('#02 - Debe mostrar el subtitulo enviado via props', () => {

        const saludo = 'Hola, soy Goku';
        const subTitulo = 'Soy un subTitulo';

        const wrapper = shallow(
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subTitulo }
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe( subTitulo );
    });
});