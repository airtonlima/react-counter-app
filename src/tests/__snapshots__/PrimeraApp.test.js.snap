// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> #01 - Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <pre>
    {
   "nombre": "Airton",
   "edad": 26
}
  </pre>
</Fragment>
`;
